instrumentation/kafka-clients/kafka-clients-0.11/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/KafkaTracing.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.kafka.KafkaHeadersGetter;
+import io.opentelemetry.instrumentation.kafka.KafkaHeadersSetter;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaTracing {
+  public static final String INSTRUMENTATION_NAME = "io.opentelemetry.kafka-clients-0.11.library";
+
+  private static final Logger logger = LoggerFactory.getLogger(KafkaTracing.class);
+
+  private static final TextMapGetter<Headers> GETTER = new KafkaHeadersGetter();
+
+  private static final TextMapSetter<Headers> SETTER = new KafkaHeadersSetter();
+
+  private final Instrumenter<ProducerRecord<?, ?>, Void> producerInstrumenter;
+  private final Instrumenter<ConsumerRecord<?, ?>, Void> consumerProcessInstrumenter;
+
+  KafkaTracing(
+      Instrumenter<ProducerRecord<?, ?>, Void> producerInstrumenter,
+      Instrumenter<ConsumerRecord<?, ?>, Void> consumerProcessInstrumenter) {
+    this.producerInstrumenter = producerInstrumenter;
+    this.consumerProcessInstrumenter = consumerProcessInstrumenter;
+  }
+
+  public static KafkaTracingBuilder create(OpenTelemetry openTelemetry) {
+    return new KafkaTracingBuilder(openTelemetry);
+  }
```

Comment:
(similar to the pattern in `OkHttpTracing`)
```suggestion
  /** Returns a new {@link KafkaTracing} configured with the given {@link OpenTelemetry}. */
  public static KafkaTracing create(OpenTelemetry openTelemetry) {
    return newBuilder(openTelemetry).build();
  }

  /** Returns a new {@link KafkaTracingBuilder} configured with the given {@link OpenTelemetry}. */
  public static KafkaTracingBuilder newBuilder(OpenTelemetry openTelemetry) {
    return new KafkaTracingBuilder(openTelemetry);
  }
```